eZ/Publish/Core/Repository/RoleService.php

```diff
@@ -1052,6 +1054,139 @@ public function unassignRoleFromUser(APIRole $role, User $user)
     }
 
     /**
+     * Removes the given role assignment.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the authenticated user is not allowed to remove a role assignment
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException If the role assignment was not found
+     *
+     * @param \eZ\Publish\API\Repository\Values\User\RoleAssignment $roleAssignment
+     */
+    public function removeRoleAssignment(RoleAssignment $roleAssignment)
+    {
+        if ($this->repository->canUser('role', 'assign', $roleAssignment) !== true) {
+            throw new UnauthorizedException('role', 'assign');
+        }
+
+        $spiRoleAssignment = $this->userHandler->loadRoleAssignment($roleAssignment->id);
+        if (!$spiRoleAssignment instanceof SPIRoleAssignment) {
```

Comment:
It shouldn't be necessary to check for this if we can rely on handler returning the object or throwing not found.
Same on 2 places below.